analyzers/tests/SonarAnalyzer.UnitTest/TestCases/CloudNative/AzureFunctionsLogFailures.cs

```diff
@@ -0,0 +1,325 @@
+//      ^^^^^
+        {
+            log.LogTrace(string.Empty);        // Secondary
+//          ^^^^^^^^^^^^^^^^^^^^^^^^^^
+            log.LogDebug(string.Empty);        // Secondary
+        }
+    }
+
+    [FunctionName("Sample")]
+    public static void LogExceptionInCatchClause(ILogger log)
+    {
+        try { }
+        catch (Exception ex) // Compliant
+        {
+            log.LogError(ex, "");
+        }
+    }
+
+    [FunctionName("Sample")]
+    public static void LogExceptionInWrappedLogger(ILogger log)
+    {
+        try { }
+        catch (Exception ex) // Compliant
+        {
+            var nullLogger = NullLogger.Instance;
+            nullLogger.Log(LogLevel.Error, new EventId(), (object)null, ex, (s, e) => string.Empty);
+        }
+    }
+
+    [FunctionName("Sample")]
+    public static void LogExceptionLoggedInLocalFunction(ILogger log)
+    {
+        try { }
+        catch (Exception ex) // FN. Any call to "log" is considered valid. Reachability is not considered.
+        {
+            void Log() => log.LogError("");
+        }
+    }
+
+    [FunctionName("Sample")]
+    public static void LogExceptionLoggedInLambda(ILogger log)
+    {
+        try { }
+        catch (Exception ex) // FN. Any call to "log" is considered valid. Reachability is not considered.
+        {
+            Action x = () => log.LogError("");
+        }
+    }
+
+    [FunctionName("Sample")]
+    public static void LogExceptionLoggedInAnonymousFunction(ILogger log)
+    {
+        try { }
+        catch (Exception ex) // FN. Any call to "log" is considered valid. Reachability is not considered.
+        {
+            Action x = delegate () { log.LogError(""); };
+        }
+    }
+
+    [FunctionName("Sample")]
+    public static void LogExceptionIsPassedInLambda(ILogger log)
+    {
+        try { }
+        catch (Exception ex) // FN. Any call to "log" is considered valid. Reachability is not considered.
+        {
+            DoSomething(() => log.LogError(""));
+        }
+
+        void DoSomething(Action a) => a();
+    }
+
+    [FunctionName("Sample")]
+    public static void LogExceptionLoggedInUnreachableCode(ILogger log)
+    {
+        try { }
+        catch (Exception ex) // FN. log in unreachable code.
+        {
+            return;
+            log.LogError("");
+        }
+    }
+
+    [FunctionName("Sample")]
+    public static void LogExceptionNotEveryCodePathContainsLog(ILogger log)
+    {
+        try { }
+        catch (Exception ex) // FN. Not every code path contains a log
+        {
+            var i = 100;
+            if (i == 42)
+            {
+                log.LogError("");
+            }
+        }
+    }
+
+    [FunctionName("Sample")]
+    public static void LogExceptionFromWebApi(ILogger log)
+    {
+        try { }
+        catch (Exception ex) // Compliant. Call to unregcognized extension method from WebApi
+        {
+            var i = 100;
+            if (i == 42)
+            {
```

Comment:
The comments above suggest, that this noise can be removed to demonstrate the idea of this test.